doc/config.rst

```diff
@@ -71,3 +71,20 @@ configure them in your config file.
         ->setIndent("\t")
         ->setLineEnding("\r\n")
     ;
+
+It's possible to register custom rule sets, which makes it easier to reuse custom configuration between multiple projects. If you have prepared rule set, you can register it and then enable it in the rules. Custom rule sets (in this example `\MyNameSpace\MyRuleSetClass`) must implement `\PhpCsFixer\RuleSet\RuleSetDescriptionInterface`.
+
+.. code-block:: php
+
+    <?php
+
+    return (new PhpCsFixer\Config())
+        ->registerCustomRuleSets([
+            '@MyRuleSet' => MyNameSpace\MyRuleSetClass::class,
+        ])
+        ->setRules([
+            '@MyRuleSet' => true,
+        ])
+    ;
+
+ℹ️ If you use other `\PhpCsFixer\ConfigInterface` implementation than built-in one, you can register custom rule sets using `\PhpCsFixer\RuleSet\RuleSets::registerRuleSet()` just before the configuration. Currently `\PhpCsFixer\Config::registerCustomRuleSets()` is not part of the `ConfigInterface` because it would be a BC-break - it will be added there in the next major version of the Fixer.
```

Comment:
do not promote calling `RuleSets::registerRuleSet` directly - it's internal class

Reply:
I believe we should have information how to use it when custom config implementation is used (and there's no API for registering ruleset). Maybe we just should explicitly state it's a part of internal API and does not have BC-promise?

Reply:
or maybe if one is using _custom ruleset_ with _custom config_, it's OK to expect them to implement future-ready Config method

Reply:
But it does not solve anything since it's not part of the contract. `Config` just works as a proxy to the static call, but in the future it won't require any changes if people stick with built-in implementation (even if it changes internal implementation to store custom rulesets to be read later in the process, rather than doing static call to `RuleSets`). Adding it in custom `ConfigInterface` implementation would only move static call from config file to that implementation - is it worth it?

Reply:
It was reworked and now it's based on dedicated contract (`CustomRulesetsAwareConfigInterface`) and static calls are not needed anymore, only config class that implements the contract.